### PR TITLE
feat: deprecate etcd meta store

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/src/controller/mod.rs
Comment (BugenZhao) [inline src/meta/src/controller/mod.rs:63]: Shall we specify a name for the instance to avoid (potentially) being conflict with other usages?

https://arc.net/l/quote/dpqhqzhz

File: src/meta/src/backup_restore/restore.rs
Comment (yezizp2012) [inline src/meta/src/backup_restore/restore.rs:196]: Cc @zwang28 , I commented out all unit tests about backup and restore, which is using `MetaSnapshotV1`. Please switch it to use sql backend after this PR get merged.

File: ci/workflows/main-cron.yml
Comment (fuyufjh) [inline ci/workflows/main-cron.yml:658]: I suppose that we will refactor the backup & restore and re-enable the backwards compatibility tests, right? cc. @zwang28

File: src/ctl/src/cmd_impl/meta/migration.rs
Comment (fuyufjh) [inline src/ctl/src/cmd_impl/meta/migration.rs:1]: Assuming a user is upgrading from 2.0 (with etcd) to 2.1, they should follow these steps: 

1. backup everything 
2. migrate to SQL using the migration utility 
3. upgrade RW cluster from 2.0 to 2.1

Is it correct?

File: proto/telemetry.proto
Comment (fuyufjh) [inline proto/telemetry.proto:10]: Tend to keep the number unchanged.
```suggestion
  META_BACKEND_RDB = 3;
```

**Review by fuyufjh — APPROVED** [review]
LGTM! Assuming the code changes are mostly unrelated to real logic, didn't read every changed file.

File: src/ctl/src/cmd_impl/meta/migration.rs
Comment (yezizp2012) [reply src/ctl/src/cmd_impl/meta/migration.rs:1]: Yes exactly.

File: src/meta/src/controller/mod.rs
Comment (yezizp2012) [reply src/meta/src/controller/mod.rs:63]: It's solely used for unit tests, and upon local testing, it appears conflict-free. Let's keep it unchanged. It should be useful in simulation tests.

File: src/meta/src/hummock/manager/compaction/mod.rs
Comment (zwang28) [inline src/meta/src/hummock/manager/compaction/mod.rs:831]: pls keep the // TODO #13952: support V2

Comment (zwang28) [inline src/meta/src/hummock/manager/compaction/mod.rs:831]: Please keep the comment: TODO #13952: support V2

**Review by zwang28 — APPROVED** [review]
LGTM for Hummock storage related

File: ci/workflows/main-cron.yml
Comment (zwang28) [reply ci/workflows/main-cron.yml:658]: Backwards compatibility test is not related to backup & restore.

Backup & restore has been testing with both etcd and sql for some time.

Comment (zwang28) [reply ci/workflows/main-cron.yml:658]: 2.0 -> 2.1 requires migration script @yezizp2012 , not backup & restore.

File: src/meta/src/hummock/manager/compaction/mod.rs
Comment (zwang28) [reply src/meta/src/hummock/manager/compaction/mod.rs:831]: I'll fix it soon.

File: ci/workflows/main-cron.yml
Comment (yezizp2012) [reply ci/workflows/main-cron.yml:658]: Currently backward compatibility tests only works for etcd backend, I will enable it once some refactoring is complete.

**Pull request merged**
